pkg/apis/core/types.go

```diff
@@ -1952,6 +1952,10 @@ const (
 	PreemptLowerPriority PreemptionPolicy = "PreemptLowerPriority"
 	// PreemptNever means that pod never preempts other pods with lower priority.
 	PreemptNever PreemptionPolicy = "Never"
+	// NonPreemptible means that pod can not be preempted by other pods with higher priority.
```

Comment:
I think it's better to open a KEP first and have a thorough discussion about why we need it and how to implement it. Please take a look at https://github.com/kubernetes/enhancements